alchemy/src/cloudflare/d1-local-migrations.ts

```diff
@@ -25,20 +25,34 @@ export const applyLocalD1Migrations = async (
     await miniflare.ready;
     const db = await miniflare.getD1Database("DB");
     const session = db.withSession("first-primary");
-    await session
-      .prepare(
-        `CREATE TABLE IF NOT EXISTS ${options.migrationsTable} (
+    const tableInfo = await session
+      .prepare(`PRAGMA table_info(${options.migrationsTable});`)
+      .all<{
+        cid: number;
+        name: string;
+        type: string;
+        notnull: number;
+        dflt_value: string | null;
+        pk: number;
+      }>();
+    if (tableInfo.results.length === 0) {
+      await session
+        .prepare(
+          `CREATE TABLE ${options.migrationsTable} (
         id INTEGER PRIMARY KEY AUTOINCREMENT,
         name TEXT NOT NULL,
-        applied_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL
+        applied_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL,
+        type TEXT NOT NULL
     )`,
-      )
-      .run();
-    await session
-      .prepare(
-        `ALTER TABLE ${options.migrationsTable} ADD COLUMN type TEXT NOT NULL DEFAULT 'migration';`,
-      )
-      .run();
+        )
+        .run();
+    } else if (!tableInfo.results.some((col) => col.name === "type")) {
+      await session
+        .prepare(
+          `ALTER TABLE ${options.migrationsTable} ADD COLUMN type TEXT NOT NULL DEFAULT 'migration';`,
+        )
+        .run();
+    }
```

Comment:
Second error cause: once the column is added, this shouldn't run again, otherwise you get a column already exists error. Checking the table info first fixes that.